### PR TITLE
dry held suarez

### PR DESCRIPTION
**Pull request by jiahe23**
This PR adds dry held suarez experiments with rho etot and rho theta as prog var for testing purposes. 

Apart from the hs forcing, it also adds the flux correction and a simple sponge layer to rhs for stabilization.

At this moment, it can run for 100 ~ over 200 days depending on the set ups. We would like to merge it to the main so that @charleskawczynski could start to improve performance while we work on stabilize it.

### CODE REVIEW

**Comment by jiahe23** [comment]
bors try

**Comment by jiahe23** [comment]
bors try

File: examples/3dsphere/held_suarez_rho_etot.jl [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+dirname =
+    "held_suarez_implicit_he-" *
+    string(helem) *
+    "_hp-" *
+    string(npoly) *
+    "_ve-" *
+    string(velem) *
+    "_days-" *
+    string(nday)
+path = joinpath(@__DIR__, "output", dirname)
```
Comment (szy21) [inline examples/3dsphere/held_suarez_rho_etot.jl:234]: I think the `dirname` here may need to be the same as in pipeline.yml

File: examples/3dsphere/held_suarez_rho_etot.jl [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+anim = Plots.@animate for sol1 in sol.u
+    uₕ = sol1.uₕ
+    uₕ_phy = Geometry.transform.(Ref(Geometry.UVAxis()), uₕ)
+    w_phy = Geometry.transform.(Ref(Geometry.WAxis()), sol1.w)
+    u = uₕ_phy.components.data.:1
+    v = uₕ_phy.components.data.:2
+    Plots.plot(u, level = 10, clim = (-5, 80))
+    # Plots.plot(v, level = 10, clim = (-6, 6))
+    # println( maximum(abs.(parent(w_phy))) )
```
Comment (szy21) [inline examples/3dsphere/held_suarez_rho_etot.jl:222]:
```suggestion

```

File: examples/3dsphere/held_suarez_rho_etot.jl [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    "_hp-" *
+    string(npoly) *
+    "_ve-" *
+    string(velem) *
+    "_days-" *
+    string(nday)
+path = joinpath(@__DIR__, "output", dirname)
+mkpath(path)
+Plots.mp4(anim, joinpath(path, "u.mp4"), fps = 5)
+# Plots.mp4(anim, joinpath(path, "v.mp4"), fps = 5)
```
Comment (szy21) [inline examples/3dsphere/held_suarez_rho_etot.jl:237]:
```suggestion

```

File: examples/3dsphere/held_suarez_rho_theta.jl [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+anim = Plots.@animate for sol1 in sol.u
+    uₕ = sol1.uₕ
+    uₕ_phy = Geometry.transform.(Ref(Geometry.UVAxis()), uₕ)
+    w_phy = Geometry.transform.(Ref(Geometry.WAxis()), sol1.w)
+    u = uₕ_phy.components.data.:1
+    v = uₕ_phy.components.data.:2
+    Plots.plot(u, level = 10, clim = (-5, 80))
+    # Plots.plot(v, level = 10, clim = (-6, 6))
+    # println( maximum(abs.(parent(w_phy))) )
```
Comment (szy21) [inline examples/3dsphere/held_suarez_rho_theta.jl:208]:
```suggestion

```

File: examples/3dsphere/held_suarez_rho_theta.jl [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    "_hp-" *
+    string(npoly) *
+    "_ve-" *
+    string(velem) *
+    "_days-" *
+    string(nday)
+path = joinpath(@__DIR__, "output", dirname)
+mkpath(path)
+Plots.mp4(anim, joinpath(path, "u.mp4"), fps = 5)
+# Plots.mp4(anim, joinpath(path, "v.mp4"), fps = 5)
```
Comment (szy21) [inline examples/3dsphere/held_suarez_rho_theta.jl:223]:
```suggestion

```

**Review by szy21 — APPROVED** [review]

**Comment by jiahe23** [comment]
bors try

**Pull request merged**